packages/driver/src/cy/commands/sessions.ts

```diff
@@ -0,0 +1,822 @@
+    },
+
+    async getCurrentSessionData () {
+      const storage = await sessions.getStorage({ origin: '*' })
+
+      let cookies = [] as any[]
+
+      cookies = await Cypress.automation('get:cookies', {})
+
+      const ses = {
+        ...storage,
+        cookies,
+      }
+
+      return ses
+    },
+
+    getSession (id) {
+      return Cypress.backend('get:session', id)
+    },
+
+    /**
+     * 1) if we only need current_origin localStorage, access sync
+     * 2) if cross-origin http, we need to load in iframe from our proxy that will intercept all http reqs at /__cypress/automation/*
+     *      and postMessage() the localStorage value to us
+     * 3) if cross-origin https, since we pass-thru https conntections in the proxy, we need to
+     *      send a message telling our proxy server to intercept the next req to the https domain,
+     *      then follow 2)
+     */
+    async getStorage (options = {}) {
+      const specWindow = cy.state('specWindow')
+
+      if (!_.isObject(options)) {
+        throw new Error('getStorage() takes an object')
+      }
+
+      const opts = _.defaults({}, options, {
+        origin: 'current_origin',
+      })
+
+      const current_origin = $Location.create(window.location.href).origin
+
+      const origins = await mapOrigins(opts.origin)
+
+      const getResults = () => {
+        return results
+      }
+      const results = {
+        localStorage: [] as any[],
+        sessionStorage: [] as any[],
+      }
+
+      function pushValue (origin, value) {
+        if (!_.isEmpty(value.localStorage)) {
+          results.localStorage.push({ origin, value: value.localStorage })
+        }
+
+        if (!_.isEmpty(value.sessionStorage)) {
+          results.sessionStorage.push({ origin, value: value.sessionStorage })
+        }
+      }
+
+      const currentOriginIndex = origins.indexOf(current_origin)
+
+      if (currentOriginIndex !== -1) {
+        origins.splice(currentOriginIndex, 1)
+        // localStorage.length propery is not always accurate, we must stringify to check for entries
+        // for ex) try setting localStorge.key = 'val' and reading localStorage.length, may be 0.
+        const _localStorageStr = JSON.stringify(window.localStorage)
+        const _localStorage = _localStorageStr.length > 2 && JSON.parse(_localStorageStr)
+        const _sessionStorageStr = JSON.stringify(window.sessionStorage)
+        const _sessionStorage = _sessionStorageStr.length > 2 && JSON.parse(JSON.stringify(window.sessionStorage))
+
+        const value = {} as any
+
+        if (_localStorage) {
+          value.localStorage = _localStorage
+        }
+
+        if (_sessionStorage) {
+          value.sessionStorage = _sessionStorage
+        }
+
+        pushValue(current_origin, value)
+      }
+
+      if (_.isEmpty(origins)) {
+        return getResults()
+      }
+
+      if (current_origin.startsWith('https:')) {
+        _.remove(origins, (v) => v.startsWith('http:'))
+      }
```

Comment:
any chance that silently excluding http origins from https origins could cause unexpected issues/confusion for users? this behavior doesn't seem to be documented in the docs PR